packages/headless/src/controllers/commerce/did-you-mean/headless-did-you-mean.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  QueryCorrection,
+  WordCorrection,
+} from '../../../api/search/search/query-corrections';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {stateKey} from '../../../app/state-key';
+import {didYouMeanReducer as didYouMean} from '../../../features/commerce/did-you-mean/did-you-mean-slice';
+import {hasQueryCorrectionSelector} from '../../../features/did-you-mean/did-you-mean-selectors';
+import {CommerceDidYouMeanSection} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+import {DidYouMeanState} from '../../did-you-mean/headless-did-you-mean';
+
+export type {QueryCorrection, WordCorrection, DidYouMeanState};
+
+export interface DidYouMean extends Controller {
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `DidYouMean` controller.
+   */
+  state: DidYouMeanState;
+}
+
+/**
+ * The DidYouMean controller is responsible for handling query corrections.
+ * When a query returns no result but finds a possible query correction, the controller automatically triggers a new
+ * query with the suggested term.
```

Comment:
```suggestion
 * When a query returns no result but a likely correction is available, the controller automatically triggers a new
 * search with the corrected query.
```

Reply:
I've removed this, since it's not true for the did you mean commerce controller, as queries are (as of now) automatically corrected.